main.sh

```diff
@@ -197,6 +197,12 @@ fi
 if [[ -n "${no_default_features}" ]]; then
     build_options+=("--no-default-features")
 fi
+manifest_path="${INPUT_MANIFEST_PATH:-}"
+if [[ -n "${manifest_path}" ]]; then
+    build_options+=("--manifest-path" "${manifest_path}")
+    build_options+=("--target-dir" "${RUNNER_TEMP}/target")
```

Comment:
It is probably preferable to use `cargo metadata` to get the default target directory, rather than setting our own target directory.

Something like the following:

```sh
if [[ -n "${manifest_path}" ]]; then
    build_options+=("--manifest-path" "${manifest_path}")
    target_dir=$(cargo metadata --format-version=1 --no-deps --manifest-path "${manifest_path}" | jq -r '."target_directory"')
else
    target_dir=$(cargo metadata --format-version=1 --no-deps | jq -r '."target_directory"')
fi
if [[ -n "${INPUT_TARGET:-}" ]]; then
    build_options+=("--target" "${target}")
    target_dir="${target_dir}/${target}/release"
else
    target_dir="${target_dir}/release"
fi
````